libflux/src/core/semantic/builtins.rs

```diff
@@ -109,7 +109,7 @@ pub fn builtins() -> Builtins<'static> {
                  "join" => "forall [t0, t1, t2] where t0: Row, t1: Row, t2: Row (left: [t0], right: [t1], fn: (left: t0, right: t1) -> t2) -> [t2]",
             },
             "generate" => semantic_map! {
-                "from" => "forall [] (start: time, stop: time, count: int, fn: (n: int) -> int) -> [{ _start: time | _stop: time | _time: time | _value:int }]",
+                "from" => "forall [t0] where t0: Timeable (start: t0, stop: t0, count: int, fn: (n: int) -> int) -> [{ _start: time | _stop: time | _time: time | _value:int }]",
```

Comment:
Unless I'm missing something, it seems like there's a little more work necessary to ensure that durations work with `generate.from`. I think that this part of the code will produce an error if you pass in durations:
https://github.com/influxdata/flux/blob/baaa4eccb94b88998b247637c9ae412e2b0d63ea/stdlib/generate/from.go#L41-L51

I know it's not feasible to write a `_test.flux` that exercises `generate.from`, but you could write a test case similar to this one for `filter`:
https://github.com/influxdata/flux/blob/be8c67e907a2d96c7f9ced8c7660fdffeb152fcc/stdlib/universe/filter_test.go#L27

That would ensure that we handle duration arguments correctly.

Reply:
> Unless I'm missing something, it seems like there's a little more work necessary to ensure that durations work with `generate.from`. I think that this part of the code will produce an error if you pass in durations:

@wolffcm I could be misunderstanding that statement, but I thought the `ToQueryTime` took care of `duration` -> `time` when the `start` or `stop` is duration. 🤔 

<img width="842" alt="Screen Shot 2020-07-06 at 8 49 27 AM" src="https://user-images.githubusercontent.com/18511823/86613266-3087d880-bf66-11ea-87da-83d8cdd3fb92.png">

seems to work: (?) 

<img width="563" alt="Screen Shot 2020-07-06 at 8 47 19 AM" src="https://user-images.githubusercontent.com/18511823/86613306-41d0e500-bf66-11ea-98a9-8497b460ab6f.png">

Reply:
Ah, ok, so it works, that's great. I didn't know the extra logic was added there. I still think there's value to a unit test, I'm not sure why there wasn't one there in the first place.